File: services/platform-java/src/main/java/com/szu/afternoon3/platform/entity/mongo/CommentDoc.java
package com.szu.afternoon3.platform.entity.mongo;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDateTime;

@Data
@Document(collection = "comments")
@CompoundIndex(name = "idx_post_time", def = "{'postId': 1, 'createdAt': -1}")
public class CommentDoc {
    @Id
    private String id;

    @Indexed
    private String postId;

    private Long userId;
    private String userNickname;
    private String userAvatar;

    private String content;

    // 父评论ID：
    // 1. 如果是回复帖子，此字段为 null (一级评论)
    // 2. 如果是回复别人，此字段固定为所属的【一级评论 ID】(为了方便聚合查询)
    @Indexed
    private String parentId;

    // 冗余字段：真正被回复的人 (方便前端显示 "回复 @某某")
    private Long replyToUserId;
    private String replyToUserNickname;

    // 【新增】回复总数 (仅一级评论维护此字段，用于前端显示 "展开X条回复")
    private Integer replyCount = 0;

    // 点赞数
    private Integer likeCount = 0;

    private LocalDateTime createdAt;
}

File: services/comment-rpc/internal/model/comment_doc.go
package model

import (
	"time"

	"comment-rpc/comment"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type CommentDoc struct {
	ID                  primitive.ObjectID `bson:"_id,omitempty"`
	PostID              string             `bson:"postId"`
	UserID              int64              `bson:"userId"`
	UserNickname        string             `bson:"userNickname"`
	UserAvatar          string             `bson:"userAvatar"`
	Content             string             `bson:"content"`
	ParentID            string             `bson:"parentId,omitempty"`
	ReplyToUserID       int64              `bson:"replyToUserId,omitempty"`
	ReplyToUserNickname string             `bson:"replyToUserNickname,omitempty"`
	ReplyCount          int32              `bson:"replyCount"`
	LikeCount           int32              `bson:"likeCount"`
	CreatedAt           time.Time          `bson:"createdAt"`
}

type CommentLikeDoc struct {
	ID        primitive.ObjectID `bson:"_id,omitempty"`
	UserID    int64              `bson:"userId"`
	CommentID string             `bson:"commentId"`
	CreatedAt time.Time          `bson:"createdAt"`
}

func (d CommentDoc) ToProto(isLiked bool) *comment.Comment {
	return &comment.Comment{
		Id:                  d.ID.Hex(),
		PostId:              d.PostID,
		UserId:              d.UserID,
		UserNickname:        d.UserNickname,
		UserAvatar:          d.UserAvatar,
		Content:             d.Content,
		ParentId:            d.ParentID,
		ReplyToUserId:       d.ReplyToUserID,
		ReplyToUserNickname: d.ReplyToUserNickname,
		ReplyCount:          d.ReplyCount,
		LikeCount:           d.LikeCount,
		IsLiked:             isLiked,
		CreatedAt:           d.CreatedAt.Unix(),
	}
}
